### ISSUE TITLE
[1.6.0.1] Issue with facebook-oauth login method when user has facebook app installed on android phone

### ISSUE THREAD

**Issue by SECTICA** [meteor/meteor#10009]
**Includes  a possible fix I implemented
**after further testing this seems to happen only on Android**

Meteor version: 1.6.0.1
accounts-facebook: 1.3.1
facebook-oauth: 1.4.1

OS: Ubuntu 16.04

We are experiencing a problem that affects a noticeable percent of our user base.
When users are requested to log-in on our website they have the option to sign-in using Facebook, which was accomplished with accounts-Facebook ( and consequently facebook-oauth ).
The issue we are experiencing is connected to the fact that some users that surf from their smart-phones have the Facebook application installed and at some point they allowed 3rd party authorization/authentication to go through the app instead of in the browser. ( “Open with…” and Facebook was selected ).

When the user taps "login with facebook" it opens the App instead of a pop-up in mobile browser, this causes the website to not recognize the authentication request/response, and throws the following error on the server:
`Error: No matching login attempt found 145546287`
As if the user closed the popup without accepting.

A link to a repository with minimal project that exhibits this behavior:
https://github.com/SECTICA/Accounts-FB-issue

To test, one needs `appId` and `secret` for Facebook login to run this, and this should be tested on a smartphone,

**Possible fix:**
I have temporarily fixed this by overriding the facebook-oauth package in the following manner,
I changed line 33 in facebook_client.js from the facebook-oauth package 

From:
`https://www.facebook.com/v3.0/dialog/oauth?client_id=' + config.appId +`
To:
`'https://' + (mobile ? 'm' : 'www') + '.facebook.com/v3.0/dialog/oauth?client_id=' + config.appId +`

For whatever reason, when a mobile device sees a `m.facebook.com/...` login link, it opens the pop-up in the mobile browser. ( as we need it to be )
But if the link is `www.facebook.com/...`, as it is in the current version of facebook-oauth, and the user has the Facebook app installed, it tries to authenticate through the app.

The fix can be seen on the same repo on a branch called "possibleFix": 
https://github.com/SECTICA/Accounts-FB-issue/tree/possibleFix

I was unable to find anyone else with this problem either because I'm phrasing my queries wrong ( non-native english speaker ) or because this issue is not wide-spread.
Opened a thread on the forum, not much activity so far.
[Forum post](https://forums.meteor.com/t/possible-fix-found-review-appreciated-fb-web-login-issue-caused-by-fb-app/44124)

Do you have any insight on this problem?

Am I missing something that should prevent the facebook app from running, or should I stick with the fix I implemented?

Also, depending on how many people are effected by this, maybe some sort of fix should be implemented in the source-code.
Maybe an option that forces `m.face...` links when on mobile?

Thank you,
Constantine

**Comment by paulincai** [comment]
Hi there,

I am on Meteor 1.7.0.1 with accounts-facebook@1.3.1
When you add account-facebook, if you check into your versions file under .meteor/local you should see a package facebook-oauth@.... version. My accounts-facebook@1.3.1 adds for me facebook-oauth@1.5.0.

I don't have a facebook-oauth record in my packages file. 

How about you meteor remove your facebook-oauth and accounts-facebook and re-add accounts-facebook.

I can confirm that I tried both web (Chrome) and mobile safari (Iphone indeed) and all ok.
I've had this setup under Meteor 1.6 too.

**Comment by SECTICA** [comment]
Hey @paulincai , thanks for the reply

a few clarifications:

> I don't have a facebook-oauth record in my packages file.

Netiher do I, I only added the version of facebook-oauth in the issue for reference, it was installed automatically along with `accounts-facebook`
Also tried it with when facebook-oauth is 1.5.0, makes no difference.

> I can confirm that I tried both web (Chrome) and mobile safari (Iphone indeed) and all ok.
> I've had this setup under Meteor 1.6 too.

I think you misunderstood the issue, might be I was not clear, I commented at the top that this seem to happen on Android phones, happens **only** when the user has the official `Facebook app` installed and selected to 'open' facebook login links with the app instead of the browser.
![screenshot_20180620-132446](https://user-images.githubusercontent.com/25430156/41653177-e5681d4e-748d-11e8-8f73-e9da7e7a77b5.png)


I have changed the title now to say android phone, 
And made my comment inside the text bold.

Also, the change I implemented seems to fix this issue, however I would like someones input about it.
